Bottom: 72c0972a96f42b084bb281a29ca81a1d26a6a1f9
Top:    ab62351715d69143bd20394e5b2b7d7aa5347faa
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-27 20:35:07 +0100

Remove hack in draw.c which broke PS (and probably other) output

The set_layer call hacked into DrawLayerGroup to reset compositing tricked
the PS exporter to re-open and overwrite output in the main layer output
file during the fab drawing output.


---

diff --git a/src/draw.c b/src/draw.c
index cc24cf9..69a06f6 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -679,8 +679,7 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
   int n_entries = PCB->LayerGroups.Number[group];
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
-  for (i = n_entries - 1; i >= 0;
-      i--, gui->set_layer (0, group, 0)) /* HACK: Subcomposite each layer in a layer group separately */
+  for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index 45c3c82..19bce70 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -442,10 +442,6 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   /* print the non-clearing polys */
   r_search (layer->polygon_tree, drawn_area, NULL, poly_callback, &info);
 
-  /* HACK: Subcomposite polygons separately from other layer primitives */
-  /* Reset the compositing */
-  gui->set_layer (0, group, 0);
-
   if (TEST_FLAG (CHECKPLANESFLAG, PCB))
     return;
